doc/sphinx/changes.rst

```diff
@@ -8,6 +8,945 @@ Recent changes
 
    .. include:: ../unreleased.rst
 
+Version 8.12
+------------
+
+Summary of changes
+~~~~~~~~~~~~~~~~~~
+
+Changes in 8.12+beta1
+~~~~~~~~~~~~~~~~~~~~~
+
+**Specification language, type inference**
```

Comment:
Given that there are 1000 lines describing the changes, how about making them easy to scan/jump to by here adding a list of hyperlinks to the subsections that follow OR making them (sub-)* sections so they appear in the TOC?

Reply:
Making them subsections doesn't make them appear in the TOC in the left-bar (too many levels) and I tried both and I prefer the appearance that the current state results in.

Reply:
Note that there will be a summary written by @mattam82 with some direct links to a selection of changelog entries.

Reply:
> Making them subsections doesn't make them appear in the TOC in the left-bar (too many levels) and I tried both and I prefer the appearance that the current state results in.

We should figure out, though not in this PR, a better way to handle this because the issue arises in other long chapters or sections.  I expect the TOC should be configurable somehow.  Manually adding hyperlinks at the beginning of the section is easy enough, but isn't self-maintaining.  There ought to be a directive to make it self maintaining.  WDYT?

Reply:
This is actually supported in Sphinx and does look great for this changelog, so I will push a new version including this.

Reply:
Looks good!

No way to thumbs-up on some comments? :-(